src/main/java/seedu/address/ui/EventListPanel.java

```diff
@@ -20,14 +20,20 @@ public class EventListPanel extends UiPart<Region> {
 
     @FXML
     private ListView<Event> eventListView;
+    private EventDetailView eventDetailView;
 
     /**
      * Creates a {@code EventListPanel} with the given {@code ObservableList}.
      */
-    public EventListPanel(ObservableList<Event> eventList) {
+    public EventListPanel(ObservableList<Event> eventList, EventDetailView eventDetailView) {
         super(FXML);
         eventListView.setItems(eventList);
         eventListView.setCellFactory(listView -> new EventListViewCell());
+        eventListView.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue != null) {
+                eventDetailView.setEvent(newValue);
+            }
+        });
```

Comment:
@michaelyql Should I change the logic in personListView to this? It seems more concise. Then I do not need to have the callback method.